electron-builder.js

```diff
@@ -21,18 +24,60 @@ const fs = require("fs");
  */
 
 const NIGHTLY_APP_ID = "im.riot.nightly";
-const NIGHTLY_APP_NAME = "element-desktop-nightly";
 const NIGHTLY_DEB_NAME = "element-nightly";
 
 const pkg = JSON.parse(fs.readFileSync("package.json", "utf8"));
 
+let buildMacOsUniversal = false;
+
 /**
  * @type {import('electron-builder').Configuration}
  * @see https://www.electron.build/configuration/configuration
  */
 const config = {
     appId: "im.riot.app",
     asarUnpack: "**/*.node",
+    beforePack: async (context) => {
+        if (context.electronPlatformName === "darwin" && context.arch === Arch.universal) {
+            buildMacOsUniversal = true;
+        }
+    },
+    afterPack: async (context) => {
+        if (context.electronPlatformName !== "darwin" || context.arch === Arch.universal || !buildMacOsUniversal) {
+            // Burn in electron fuses, for macOS if we are building a universal package we only need to burn fuses there
+            const ext = {
+                darwin: ".app",
+                win32: ".exe",
+                linux: [""],
+            }[context.electronPlatformName];
+
+            let executableName = context.packager.appInfo.productFilename;
+            if (context.electronPlatformName === "linux") {
+                // Linux uses the package name as the executable name
+                executableName = context.packager.appInfo.name;
+            }
+
+            const electronBinaryPath = path.join(context.appOutDir, `${executableName}${ext}`);
+            console.log("Flipping fuses for: ", electronBinaryPath);
+
+            await flipFuses(electronBinaryPath, {
+                version: FuseVersion.V1,
+                resetAdHocDarwinSignature: context.electronPlatformName === "darwin" && context.arch === Arch.universal,
+
+                [FuseV1Options.EnableCookieEncryption]: true,
+                [FuseV1Options.OnlyLoadAppFromAsar]: true,
+
+                [FuseV1Options.RunAsNode]: false,
+                [FuseV1Options.EnableNodeOptionsEnvironmentVariable]: false,
+                [FuseV1Options.EnableNodeCliInspectArguments]: false,
+
+                // Mac app crashes when enabled for us on arm, might be fine for you
```

Comment:
You might think it obvious, but when I read this, I wondered "when *what* is enabled?"  At first I thought I thought you meant the whole mac app.

```suggestion
                // Mac app crashes on arm for us when `LoadBrowserProcessSpecificV8Snapshot` is enabled; might be fine for you
```